### PR TITLE
Update WB line headers when changing mapping field

### LINKED ISSUE #1858: WorkBench mapper not saving assigned determination number
chrome_88Us8P2sTu

In my data set, I mapped the field text8 to the #3 Determination and it automatically reset to the #2 Determination after saving.

version: 7.7.0
browser: Chrome

There were 3 mapped determinations that included other fields.

Both the mapping and data set are here:
[WVA 0-10000 (7-18 Export).zip](https://github.com/specify/specify7/files/9144601/WVA.0-10000.7-18.Export.zip)

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #1858

### Checklist

- [ ] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [ ] Add relevant issue to release milestone

### Testing instructions
TODO

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts [hunk 1]
```diff
@@ -251,12 +255,21 @@ export const reducer = generateReducer<MappingState, MappingActions>({
     )
       return state;
 
+    const newLine = replaceItem(state.lines, focusedLine, {
+      ...state.lines[focusedLine],
+      mappingPath: mappingViewMappingPath,
+    });
+
+    const newHeader = generateMappingPathPreview(
+      baseTableName as keyof Tables,
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts:264]: you should be able to get the base table name from `state`

File: specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts [hunk 2]
```diff
@@ -251,12 +255,21 @@ export const reducer = generateReducer<MappingState, MappingActions>({
...
+      newLine[focusedLine].mappingPath
+    );
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts:266]: hmm, doesn't look right
the `headerName` is supposed to be coming from the data set
the only time you would be setting header name to result of `generateMappingPathPreview()` is when you are trying to add a new column to the data set - which doesn't seem to be the case here?

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts (hunk 2 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts:266]: we need to reset the header when the user changes the fields in an existing line

**Comment by CarolineDenis** [comment]
before: 

https://github.com/user-attachments/assets/9c56ff54-bc76-4412-85e2-7e42dc0d8d79

now: 

https://github.com/user-attachments/assets/16d8f75c-b578-4b21-8bf5-797af704fb78

File: specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts (hunk 1 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts:264]: not sure, how would you? 
export type MappingState = State<
  'MappingState',
  {
    readonly showMappingView: boolean;
    readonly showHiddenFields: boolean;
    readonly mappingView: MappingPath;
    readonly mappingsAreValidated: boolean;
    readonly lines: RA<MappingLine>;
    readonly focusedLine: number;
    readonly changesMade: boolean;
    readonly mustMatchPreferences: IR<boolean>;
    readonly autoMapperSuggestions?: RA<AutoMapperSuggestion>;
    readonly openSelectElement?: SelectElementPosition;
    readonly validationResults: RA<MappingPath>;
  }
>;

Comment (maxpatiiuk) [reply specifyweb/frontend/js_src/lib/components/WbPlanView/mappingReducer.ts:264]: > not sure, how would you?

you are right!
sorry, I forgot that it is coming as a prop, and not part of the state

**Comment by maxpatiiuk** [comment]
This is wrong behavior
Headers are defined by the user in the CSV - we do not touch them.
See:

https://github.com/user-attachments/assets/42d3b950-8507-4e0e-aea7-4c7abb010041


The ONLY headers that we touch are the headers for newly added columns.
BUT, we are already renaming those, so there is no need to do that again. See:
https://github.com/specify/specify7/blob/6e3bbb30dc8a3a43a327a7adfa04ecc1d02708b4/specifyweb/frontend/js_src/lib/components/WbPlanView/helpers.ts#L51-L55

**Pull request closed without merging**